src/data_structures/linked_list.jl

```diff
@@ -0,0 +1,178 @@
+
+        current_node = current_node.next
+    end
+
+    return root_node
+end
+
+# Add a new node to the list at the specified index
+function insert(list::Node, new_node::Node, index::Int=1)
+    current_node = get_node(list, index-1)
+
+    next = current_node.next
+    current_node.next = new_node
+    new_node.next = next
+end
+
+# Create a new node with value 'val' in the list at the specified index
+function insert(list::Node, val, index::Int=1)
+    new_node = create_node(val)
+    insert(list, new_node, index)
+end
+
+# Add a new node to the end of the list
+function push_back(list::Node, node::Node)
+    current_node = list
+    while !ismissing(current_node.next)
+        current_node = current_node.next
+    end
+    
+    current_node.next = node
+end
+
+# Add a new node with value 'val' to the end of the list
+function push_back(list::Node, val)
+    new_node = create_node(val)
+    push_back(list, new_node)
+end
+
+function return_as_array(list::Node)
+    arr = Array{Any}(missing, 0)
+    node = list.next
+
+    while !ismissing(node)
+        push!(arr, node.val)
+        node = node.next
+    end
+
+    return arr
+end
+
+# Remove all elements from the list
+function clear(list::Node)
+    list.next = missing
+end
+
+# Remove an element at the specified index
+function remove(list::Node, index::Int)
+    node_before = get_node(list, index-1)
+    to_remove = node_before.next
+    node_before.next = to_remove.next
+end
+
+# Remove all elements with the value 'val'
+function remove_all(list::Node, val)
+    current_node = list
+    while !ismissing(current_node.next)
+        next_node = current_node.next
+        if next_node.val == val
+            current_node.next = next_node.next
+        else
+            current_node = next_node
+        end
+    end
+end
+
+# Remove the first element in the list with value 'val'
+function remove_first(list::Node, val)
+    index = indexOf(list, val)
+    remove(list, index)
+end
+
+# Get a node at the specified index
+function get_node(list::Node, index::Int)
+    current_node = list
+    for i in 1:index
+        if !ismissing(current_node)
+            current_node = current_node.next
+        else
+            return missing
+        end
+    end
+    return current_node
+end
+
+# Get a value from the node at the specified index
+function get(list::Node, index::Int)
+    node = get_node(list, index)
+    if !ismissing(node) 
+        return node.val
+    end
+    return missing
+end
+
+# Return the index of the first element with a value 'val'
+function indexOf(list::Node, val)
+    current_node = list.next
+    i=1
+    while !ismissing(current_node)
+        if current_node.val == val
+            return i
+        else
+            current_node = current_node.next
+            i+=1
+        end
+    end
+    return -1
+end
+
+# Return true if list is empty, false if it has elements
+function is_empty(list)
+    return ismissing(list.next)
+end
+
+end
```

Comment:
Missing an endline. Without this endline, there could be issues with some IDEs and configurations.

```suggestion
end

```